### ISSUE TITLE
[BUG] TraceDataset fails to convert Span dataframe to Parquet due to `document.metadata` being an empty `dict`

### ISSUE THREAD

**Issue by RogerHYang** [Arize-ai/phoenix#1786]
Use following code to reproduce, note `run_in_thread=False`
```python
import phoenix as px

ds = px.load_example_traces("llama_index_rag")
px.launch_app(trace=ds, run_in_thread=False)
```

<img width="757" alt="Screenshot 2023-11-20 at 1 40 57 PM" src="https://github.com/Arize-ai/phoenix/assets/80478925/8adbc15e-b0f3-49ec-9744-e63e9672b108">

<img width="873" alt="Screenshot 2023-11-20 at 1 39 55 PM" src="https://github.com/Arize-ai/phoenix/assets/80478925/38224dd4-40c3-428c-8822-7da3880e6a99">

<img width="508" alt="Screenshot 2023-11-20 at 1 44 05 PM" src="https://github.com/Arize-ai/phoenix/assets/80478925/5a1f4ad4-0096-494b-b031-d2fddb4f7749">

**Comment by mikeldking** [comment]
Possible duplicate of #1758

**Comment by RogerHYang** [comment]
> Possible duplicate of #1758

they're different: that one has to with incompatible object types for the protobuf, while this issue is a problem is with Arrow serialization

**Comment by mikeldking** [comment]
fixed via #1798
